lib/chef/provider/package/macports.rb

```diff
@@ -3,7 +3,7 @@ class Provider
     class Package
       class Macports < Chef::Provider::Package
 
-        provides :macports_package, os: "mac_os_x"
+        provides :macports_package, os: "darwin"
```

Comment:
I could see arguments either way.
For consistency's sake maybe it would be best to remove the filters. (see homebrew @ https://github.com/opscode/chef/blob/master/lib/chef/provider/package/homebrew.rb#L29)
And then if we want to unload the classes later we could do that in a separate PR.